### PR TITLE
Add web-based OAuth authentication

### PR DESCRIPTION
**Pull request by brendandburns**
Also add /approve to the actions

Builds off of #1908 so merge that first.

### CODE REVIEW

File: mungegithub/mungers/bulk-lgtm.go [hunk 1]
```diff
@@ -17,12 +17,19 @@ limitations under the License.
...
+	"encoding/binary"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 	"strconv"
+	"strings"
 	"sync"
 
+	"golang.org/x/oauth2"
```
Comment (krzyzacy) [inline mungegithub/mungers/bulk-lgtm.go]: you want to vendor the lib :-)

Comment (brendandburns) [inline mungegithub/mungers/bulk-lgtm.go]: It's already there afaik:

https://github.com/kubernetes/test-infra/tree/master/vendor/golang.org/x/oauth2

Comment (krzyzacy) [inline mungegithub/mungers/bulk-lgtm.go]: you are 100% right..

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

File: mungegithub/mungers/bulk-lgtm.go (hunk 1 above)
Comment (krzyzacy) [inline mungegithub/mungers/bulk-lgtm.go]: oh, yup... you need to add oauth2 to the [BUILD file](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/BUILD#L90)

**Comment by apelisse** [comment]
Looks good to me.

Should we remove the part about approval though, because of https://github.com/kubernetes/test-infra/issues/1744

**Comment by brendandburns** [comment]
@apelisse anything more needed here? looks like Travis is stuck...

**Comment by apelisse** [comment]
Yeah travis is fine, it completed successfully, just didn't update the status.
Could you remove the part about approval, as LGTM now auto-approves?

**Comment by apelisse** [comment]
@brendanburns

**Comment by apelisse** [comment]
@brendandburns

**Comment by brendandburns** [comment]
@apelisse removed `/approve`, added `/release-note-none`

Also moved to storing the oauth token in a cookie.

Please take another look.

Thanks!
--brendan

**Comment by apelisse** [comment]
Sorry for not seeing that earlier!

/lgtm

**Comment by brendandburns** [comment]
@apelisse what is needed to merge this?

Thanks!

**Comment by apelisse** [comment]
Nothing. test-infra is on a manual merge basis...

**Pull request merged**